js/data_v4.js

```diff
@@ -0,0 +1,382 @@
+var ONE_DAY = 1000 * 60 * 60 * 24;
+var ONE_WEEK = ONE_DAY * 7;
+var TWO_WEEKS = ONE_DAY * 14;
+var THIRTY_DAYS_IN_SECONDS = 2592000;
+var PAINT_TIME_THRESHOLD = 300000;
+var CONFIG_URL = 'js/config.json?';
+var prefs = null;
+
+Promise.prototype.finally = function(fn) {
+    return this.then(
+        function(v) {
+            fn();
+            return v;
+        },
+        function(err) {
+            fn();
+            throw err;
+        }
+    );
+};
+
+var dateFormat = new Intl.DateTimeFormat(undefined, { day: 'numeric', month: 'short', year: 'numeric' });
+function displayDate(timestamp) {
+    return dateFormat.format(new Date(timestamp));
+}
+
+// Is this the first load for the document?
+var isFirstLoad = true;
+
+// Converts the date passed to a Date object and checks
+// whether the current month is equal to the month of the
+// day argument passed.
+var isCurrentMonth = function(day, now) {
+    if (!now) {
+        now = new Date();
+    }
+
+    var currentYear = new Date().getYear();
+    var currentMonth = new Date().getMonth() + 1;
+    var year = new Date(day).getYear();
+    var month = new Date(day).getMonth() + 1;
+
+    if (currentYear === year && currentMonth === month) {
+        return true;
+    }
+    return false;
+};
+
+function isPastNDays(day, n, now) {
+    if (!now) {
+        now = new Date();
+    }
+
+    var difference = new Date().getTime() - new Date(day).getTime();
+    return difference < ONE_DAY * n;
+}
+
+function populateEnvironment(environment) {
+    var vitalStatsValueContainers = $('#vital_stats .statsBoxSection-value');
+
+    var channel = environment.settings.update.channel;
+    var updates = environment.settings.update.enabled ?
+      (environment.settings.update.autoDownload ? "automatic" : "prompt") : "disabled";
```

Comment:
nested multi-line ternaries make this unnecessarily difficult to read. recommend rewriting these as expanded if/else